source/Conversation.cpp

```diff
@@ -566,30 +568,32 @@ bool Conversation::LoadDestinations(const DataNode &node, const ConditionsStore
 	bool hasCondition = false;
 	for(const DataNode &child : node)
 	{
-		if(child.Size() == 2 && child.Token(0) == "goto" && hasGoto)
+		const string &key = child.Token(0);
+		bool hasValue = child.Size() >= 2;
+		if(key == "goto" && hasValue && hasGoto)
 		{
 			child.PrintTrace("Warning: Ignoring extra endpoint in conversation choice:");
 		}
```

Comment:
The nested `if` like you did with the `to display` looks better (see my next suggestion).
~~of course github won't let me make suggestions over deleted lines~~
```suggestion
```